polkadot/node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -758,6 +804,8 @@ async fn advertise_collation<Context>(
 
 		per_relay_parent
 			.validator_group
+			.get_mut(&core_index)
+			.expect("should_advertise returned `ShouldAdvertiseTo::Yes` above; qed")
```

Comment:
Expect is "fine" here, although there is a little too much code in between for my taste. Maybe we can do better if we bail on `None` above?